odiglet/pkg/metrics/metrics.go

```diff
@@ -0,0 +1,169 @@
+package metrics
+
+import (
+	"context"
+
+	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	otelmetric "go.opentelemetry.io/otel/metric"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	otelMeterName   = "github.com/odigos.io/odigos/odiglet"
+	languageAttrKey = "language"
+)
+
+var meter = otel.Meter(otelMeterName)
+
+// OdigletMetrics tracks unified metrics for all instrumented pods
+type OdigletMetrics struct {
+	k8sClient client.Client
+
+	// instrumentedPodsByLanguage tracks all instrumented pods per language
+	instrumentedPodsByLanguage otelmetric.Int64ObservableGauge
+
+	registration otelmetric.Registration
+}
+
+// NewOdigletMetrics creates unified metrics based on InstrumentationConfig
+func NewOdigletMetrics(k8sClient client.Client) (*OdigletMetrics, error) {
+	m := &OdigletMetrics{
+		k8sClient: k8sClient,
+	}
+
+	var err error
+	m.instrumentedPodsByLanguage, err = meter.Int64ObservableGauge(
+		"odigos.odiglet.instrumented_pods_by_language",
+		otelmetric.WithDescription("Number of instrumented pods per programming language"),
+		otelmetric.WithUnit("{pod}"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Register callback to observe the gauge
+	m.registration, err = meter.RegisterCallback(
+		m.observeInstrumentedPods,
+		m.instrumentedPodsByLanguage,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+// observeInstrumentedPods counts instrumented pods by language using InstrumentationConfig
+func (m *OdigletMetrics) observeInstrumentedPods(ctx context.Context, observer otelmetric.Observer) error {
+	counts := make(map[string]int)
+
+	// List all pods (the cache is already filtered to this node via manager config)
+	var podList corev1.PodList
+	if err := m.k8sClient.List(ctx, &podList); err != nil {
+		return nil
+	}
+
+	// For each pod, check if it has an associated InstrumentationConfig with enabled agents
+	for _, pod := range podList.Items {
+		if pod.Status.Phase != corev1.PodRunning {
+			continue
+		}
```

Comment:
This will also include pods where odigos was not actually injected or agent not healthy (might be ok).

Checking the pod Phase can be tricky

Reply:
I think we'll leave it at this for now, as this is used mostly for the graph. In the future we'll add support for this, when our state machine will be less complex